src/app/suggestions-page/suggestions-page.component.ts

```diff
@@ -0,0 +1,286 @@
+
+  /**
+   * A list of remote data objects of suggestions
+   */
+  suggestionsRD$: BehaviorSubject<PaginatedList<Suggestion>> = new BehaviorSubject<PaginatedList<Suggestion>>({} as any);
+
+  targetRD$: Observable<RemoteData<SuggestionTarget>>;
+  targetId$: Observable<string>;
+
+  suggestionTarget: SuggestionTarget;
+  suggestionId: any;
+  suggestionSource: any;
+  researcherName: any;
+  researcherUuid: any;
+
+  selectedSuggestions: { [id: string]: Suggestion } = {};
+  isBulkOperationPending = false;
+
+  constructor(
+    private authService: AuthService,
+    private notificationService: NotificationsService,
+    private paginationService: PaginationService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private suggestionService: SuggestionsService,
+    private suggestionTargetsStateService: SuggestionTargetsStateService,
+    private translateService: TranslateService,
+    private workspaceItemService: WorkspaceitemDataService
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.targetRD$ = this.route.data.pipe(
+      map((data: Data) => data.suggestionTargets as RemoteData<SuggestionTarget>),
+      redirectOn4xx(this.router, this.authService)
+    );
+
+    this.targetId$ = this.targetRD$.pipe(
+      getFirstSucceededRemoteDataPayload(),
+      map((target: SuggestionTarget) => target.id)
+    );
+    this.targetRD$.pipe(
+      getFirstSucceededRemoteDataPayload()
+    ).subscribe((suggestionTarget: SuggestionTarget) => {
+      this.suggestionTarget = suggestionTarget;
+      this.suggestionId = suggestionTarget.id;
+      this.researcherName = suggestionTarget.display;
+      this.suggestionSource = suggestionTarget.source;
+      this.researcherUuid = this.suggestionService.getTargetUuid(suggestionTarget);
+      this.updatePage();
+    });
+
+    this.suggestionTargetsStateService.dispatchMarkUserSuggestionsAsVisitedAction();
+  }
+
+  /**
+   * Called when one of the pagination settings is changed
+   */
+  onPaginationChange() {
+    this.updatePage();
+  }
+
+  /**
+   * Update the list of suggestions
+   */
+  updatePage() {
+    this.processing$.next(true);
+    const pageConfig$: Observable<FindListOptions> = this.paginationService.getFindListOptions(
+      this.paginationOptions.id,
+      this.defaultConfig,
+    ).pipe(
+      distinctUntilChanged()
+    );
+    combineLatest([this.targetId$, pageConfig$]).pipe(
+      switchMap(([targetId, config]: [string, FindListOptions]) => {
+        return this.suggestionService.getSuggestions(
+          targetId,
+          config.elementsPerPage,
+          config.currentPage,
+          config.sort
+        );
+      }),
+      take(1)
+    ).subscribe((results: PaginatedList<Suggestion>) => {
+      this.processing$.next(false);
+      this.suggestionsRD$.next(results);
+      this.suggestionService.clearSuggestionRequests();
+      // navigate to the mydspace if no suggestions remains
+
+      // if (results.totalElements === 0) {
```

Comment:
There's a lot of commented out code here. If it's needed we should uncomment. If not, it should be removed.